### DISCUSSION TITLE
automatically reboot hanging ISM

### DISCUSSION THREAD

**Discussion by Andy1616** [zivillian/ism7mqtt#146] [Ideas]
Hi
running the tool and falling in love with I need to realize that the ISM seems to be in hanging state every now and then.
For me that happens like once a day...

IDEA: 
What if ism7mqtt can detect that and send the ism a reboot cmd??

curl -vv -u admin:thepassword -d "name=reboKickoff&rebo=init" -X POST "http://ism.ip.put.here/protect/reboot.htm"

solves the issue for me...

**Comment by Eba-M** [comment]
i run the ism7mqtt within screen, using a script, it restart ism7mqtt after it stops.

**Comment by Andy1616** [comment]
the need is not to retart ism7mqtt BUT to restart ism7 module in case that one started hanging

**Comment by drose28357** [comment]
How do you send this curl comand ? Did you use a config.yaml section ?
Here is a discussion how to integrate curl into Actions 
https://community.home-assistant.io/t/execute-curl-command-from-an-automation/525604/8

**Reply by Andy1616** [reply]
Hi - well, why to bother HA if ism7mqtt can discover and act? It already has all the needed infromation, URLs and credentials...
(in my env in addition there is no HA avail...)

**Comment by drose28357** [comment]
I would like to have this automation in ism7mqtt. But it isn't there. When my ism7 stalls i will try your reboot curl in the HA shell.

**Comment by drose28357** [comment]
Today my ISM7 stalled. I used your curl on Windows 11 at the CMD shell. Works flawles ...
Thank you very much !

`Microsoft Windows [Version 10.0.26200.7462]
(c) Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\ditma>curl -vv -u admin:u..................Q -d "name=reboKickoff&rebo=init" -X POST "http://192.168.101.58/protect/reboot.htm"

Note: Unnecessary use of -X or --request, POST is already inferred.
21:33:40.493000 [0-0] * [SETUP] added
21:33:40.496000 [0-0] *   Trying 192.168.101.58:80...
21:33:40.500000 [0-0] * [SETUP] Curl_conn_connect(block=0) -> 0, done=0
21:33:40.504000 [0-0] * [SETUP] Curl_conn_connect(block=0) -> 0, done=1
21:33:40.506000 [0-0] * Established connection to 192.168.101.58 (192.168.101.58 port 80) from 192.168.101.53 port 61253
21:33:40.509000 [0-0] * [SETUP] query ALPN
21:33:40.512000 [0-0] * using HTTP/1.x
21:33:40.514000 [0-0] * Server auth using Basic with user 'admin'
21:33:40.516000 [0-0] > POST /protect/reboot.htm HTTP/1.1
21:33:40.516000 [0-0] > Host: 192.168.101.58
21:33:40.516000 [0-0] > Authorization: Basic Y...............................R
21:33:40.516000 [0-0] > User-Agent: curl/8.16.0
21:33:40.516000 [0-0] > Accept: */*
21:33:40.516000 [0-0] > Content-Length: 26
21:33:40.516000 [0-0] > Content-Type: application/x-www-form-urlencoded
21:33:40.516000 [0-0] >
21:33:40.533000 [0-0] * upload completely sent off: 26 bytes
21:34:40.586000 [0-0] * Recv failure: Connection was reset
21:34:40.589000 [0-0] * closing connection #0
curl: (56) Recv failure: Connection was reset

C:\Users\ditma>ping 192.168.101.58

Ping wird ausgeführt für 192.168.101.58 mit 32 Bytes Daten:
Antwort von 192.168.101.58: Bytes=32 Zeit=1ms TTL=64
`

**Comment by drose28357** [comment]
Hab meine Home Assistant-Lösung hier beschrieben und dieses Issue dort zitiert.. 

[github.com/b3nn0/hassio-addon-ism7mqtt/issues/78#issuecomment-3707963619](https://github.com/b3nn0/hassio-addon-ism7mqtt/issues/78#issuecomment-3707963619)

Danke für den Curl !!!
